lib/authentication.js

```diff
@@ -0,0 +1,172 @@
+'use strict';
+
+const MAX_PASS_LENGTH = 128;
+const MIN_PASS_LENGTH = 10;
+const MIN_PHRASE_LENGTH = 20;
+
+const MIN_LOGIN_LENGTH = 6;
+const MAX_LOGIN_LENGTH = 255;
+
+const passwordTests = {
+  MIN_LENGTH: [
+    password => password.length < MIN_PASS_LENGTH,
+    `The password must be longer than ${MIN_PASS_LENGTH} characters`
+  ],
+  MAX_LENGTH: [
+    password => password.length > MAX_PASS_LENGTH,
+    `The password must be less than ${MAX_PASS_LENGTH} characters`
+  ],
+  MIN_PHRASE_LENGTH: [
+    password => password.length < MIN_PHRASE_LENGTH,
+    `The passphrase must be longer than ${MIN_PHRASE_LENGTH} characters`
+  ],
+  REPEAT_CHARS: [
+    password => /(.)\1{2,}/.test(password),
+    'The password can not contain a sequence of repeated chars'
+  ],
+  ONE_LOWERCASE_CHAR: [
+    password => !/[a-z]/.test(password),
+    'The password must contain lowercase char'
+  ],
+  ONE_UPPERERCASE_CHAR: [
+    password => !/[A-Z]/.test(password),
+    'The password must contain uppercase char'
+  ],
+  ONE_NUMBER: [
+    password => !/[0-9]/.test(password),
+    'The password must contain number'
+  ],
+  ONE_SPECIAL_CHAR: [
+    password => !/[^A-Za-z0-9]/.test(password),
+    'The password must contain special char'
+  ]
```

Comment:
All of the `ONE` tests should probably be generalized to `N` chars, wdyt?

Reply:
Firstly, our tests do not have input parameters and I do not quite understand how you want to set the n. Secondly, the specifications specify tests for one character.